test/unittest/tests/wrappers/mock_session.h

```diff
@@ -27,6 +44,17 @@ class MockSession {
         return _mockConnection;
     };
 
+    void
+    add_callback_message(const char *message)
+    {
+        _messages.push_back(std::string(message));
```

Comment:
I am not an expert at C++, so ignore me if I am talking nonsense. Could we utilise streams in this case? If not, I was thinking a queue would fit exactly what you want here right?

Reply:
We could used streams here, but it's probably not necessary as the messages are simply `const char *` strings.
A queue could work, but I think the more flexible std::list is a better choice here (see my comment below)

Reply:
Yeah I'm not aiming for a performant solution as we are in unit test land.